### DISCUSSION TITLE
Custom Types via TryFrom, TryInto

### DISCUSSION THREAD

**Discussion by SZenglein** [SeaQL/sea-orm#2237] [Ideas]
[New type](https://www.sea-ql.org/SeaORM/docs/next/generate-entity/newtype/) enables usage of basic wrapper types and JSON support. 

However, there are some cases where database columns can be better represented by structs. In addition, (de-)serializing using structs has the advantage of being able to fail early. A good example is the `FromJsonQueryResult` derive, where I can kind of shift the deserialization error to sea-orm.

I think it might be best explained using serde:  

The `DeriveValueType` macro behaves the same as `#[serde(transparent)] `, 

Serde also  has [try_from](https://serde.rs/container-attrs.html#try_from). Sea-orm could first read as the respective database type and then use try_from to fallibly convert to the target type. The same applies in the reverse direction.

This can be implemented manually already, but requires 4 traits as specified in the docs, which is a lot of boilerplate.

**Comment by angrynode** [comment]
For stringy types, that's now supported on master branch. [See docs](https://www.sea-ql.org/SeaORM/docs/generate-entity/newtype/#string-like-structs)

**Comment by tyt2y3** [comment]
thanks @angrynode for raising this thread, it's now released in the latest rc
